### ISSUE TITLE
感谢作者的工作！训练自己的数据集仍旧会出现loss为0的情况！

### ISSUE THREAD

**Issue by liugaolian** [DetectionTeamUCAS/RetinaNet_Tensorflow_Rotation#23]
感谢作者，前几天更新了代码，目前训练我的数据集loss不会出现Nan了，但是依旧会出现loss为0的情况，如下所示：
************************************************************
[2019-07-31 19:41:10] global_step:1035  current_step:1035  per_cost_time:1.036s
cls_loss:1.436    reg_loss:0.588    total_losses:2.024    
************************************************************
[2019-07-31 19:41:15] global_step:1040  current_step:1040  per_cost_time:1.011s
cls_loss:1.184    reg_loss:0.614    total_losses:1.798    
************************************************************
[2019-07-31 19:41:20] global_step:1045  current_step:1045  per_cost_time:1.066s
cls_loss:0.467    **reg_loss:0.000**    total_losses:0.467    
************************************************************
[2019-07-31 19:41:26] global_step:1050  current_step:1050  per_cost_time:1.079s
cls_loss:1.418    reg_loss:0.469    total_losses:1.887    
************************************************************
[2019-07-31 19:41:31] global_step:1055  current_step:1055  per_cost_time:1.070s
cls_loss:1.206    reg_loss:0.575    total_losses:1.781    
************************************************************
[2019-07-31 19:41:36] global_step:1060  current_step:1060  per_cost_time:1.025s
cls_loss:1.335    reg_loss:0.463    total_losses:1.798    
************************************************************
[2019-07-31 19:41:42] global_step:1065  current_step:1065  per_cost_time:1.039s
cls_loss:1.209    reg_loss:0.550    total_losses:1.759    
************************************************************
[2019-07-31 19:41:47] global_step:1070  current_step:1070  per_cost_time:0.951s
cls_loss:1.000    reg_loss:0.512    total_losses:1.512    
************************************************************
[2019-07-31 19:41:52] global_step:1075  current_step:1075  per_cost_time:1.074s
cls_loss:0.891    reg_loss:0.636    total_losses:1.526

**Comment by yangxue0827** [comment]
你的数据量多少？ @liugaolian

**Comment by yangxue0827** [comment]
初期出现0挺正常的，说明那张图片你没有匹配到正anchor @liugaolian

**Comment by liugaolian** [comment]
> 你的数据量多少？ @liugaolian

嗯，我的训练图片是720张，想要检测图书书脊（书脊跟你之前论文提到的舰船检测差不多，都是旋转目标而且是细长型目标，所以我就想直接用你的模型，谢谢），但是很奇怪，竟然振荡了，目标也没有检测出来（用你的R2CNN检测效果还不错），如下图所示：


![选区_001](https://user-images.githubusercontent.com/28881571/62260620-2f405880-b445-11e9-906f-4fb586955ae9.png)
![选区_002](https://user-images.githubusercontent.com/28881571/62260629-34050c80-b445-11e9-9c67-9fdf18711a07.png)


下面是我的cfgs.py : 
#--------------------------------------------------------------------------------------------------------------------------------
# -*- coding: utf-8 -*-
from __future__ import division, print_function, absolute_import
import os
import tensorflow as tf
import math

# ------------------------------------------------
VERSION = 'bookSpine_RetinaNet'   # RetinaNet_DOTA_1x_20190606
NET_NAME = 'resnet101_v1d'  
ADD_BOX_IN_TENSORBOARD = True

# ---------------------------------------- System_config
ROOT_PATH = os.path.abspath('../')
print(20*"++--")
print(ROOT_PATH)
GPU_GROUP = "1"    #  支持多GPU训练，比如"0,1,2,3,4,5,6,7"
NUM_GPU = len(GPU_GROUP.strip().split(','))
SHOW_TRAIN_INFO_INTE = 5
SMRY_ITER = 500
SAVE_WEIGHTS_INTE = 2000

SUMMARY_PATH = ROOT_PATH + '/output/summary'
TEST_SAVE_PATH = ROOT_PATH + '/tools/test_result'

if NET_NAME.startswith("resnet"):
    weights_name = NET_NAME
elif NET_NAME.startswith("MobilenetV2"):
    weights_name = "mobilenet/mobilenet_v2_1.0_224"
else:
    raise Exception('net name must in [resnet_v1_101, resnet_v1_50, MobilenetV2]')

PRETRAINED_CKPT = ROOT_PATH + '/data/pretrained_weights/' + weights_name + '.ckpt'
TRAINED_CKPT = os.path.join(ROOT_PATH, 'output/trained_weights')
EVALUATE_DIR = ROOT_PATH + '/output/evaluate_result_pickle/'

# ------------------------------------------ Train config
RESTORE_FROM_RPN = False
FIXED_BLOCKS = 1  # allow 0~3
FREEZE_BLOCKS = [True, False, False, False, False]  # for gluoncv backbone
USE_07_METRIC = True

MUTILPY_BIAS_GRADIENT = 2.0  # if None, will not multipy
GRADIENT_CLIPPING_BY_NORM = 10.0  # if None, will not clip

CLS_WEIGHT = 1.0
REG_WEIGHT = 1.0
USE_IOU_FACTOR = True

BATCH_SIZE = 1
EPSILON = 1e-5
MOMENTUM = 0.9
LR = 5e-4  # * NUM_GPU * BATCH_SIZE
DECAY_STEP = [SAVE_WEIGHTS_INTE*12, SAVE_WEIGHTS_INTE*16, SAVE_WEIGHTS_INTE*20]
MAX_ITERATION = SAVE_WEIGHTS_INTE*20       # 总迭代次数
WARM_SETP = int(1.0 / 8.0 * SAVE_WEIGHTS_INTE)

# -------------------------------------------- Data_preprocess_config
DATASET_NAME = 'pascal'  # 'pascal', 'coco'
PIXEL_MEAN = [123.68, 116.779, 103.939]  # R, G, B. In tf, channel is RGB. In openCV, channel is BGR
PIXEL_MEAN_ = [0.485, 0.456, 0.406]
PIXEL_STD = [0.229, 0.224, 0.225]  # R, G, B. In tf, channel is RGB. In openCV, channel is BGR
IMG_SHORT_SIDE_LEN = 800
IMG_MAX_LENGTH = 1088
CLASS_NUM = 1  # 只有1类需要检测：书脊
# --------------------------------------------- Network_config
SUBNETS_WEIGHTS_INITIALIZER = tf.random_normal_initializer(mean=0.0, stddev=0.01, seed=None)
SUBNETS_BIAS_INITIALIZER = tf.constant_initializer(value=0.0)
PROBABILITY = 0.01
FINAL_CONV_BIAS_INITIALIZER = tf.constant_initializer(value=-math.log((1.0 - PROBABILITY) / PROBABILITY))
WEIGHT_DECAY = 1e-4
# ---------------------------------------------Anchor config
LEVEL = ['P3', 'P4', 'P5', 'P6', 'P7']
BASE_ANCHOR_SIZE_LIST = [32, 64, 128, 256, 512]
ANCHOR_STRIDE = [8, 16, 32, 64, 128]

ANCHOR_SCALES = [2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)]
ANCHOR_RATIOS = [1, 1 / 3., 3., 5., 1 / 5.]

ANCHOR_ANGLES = [-90, -75, -60, -45, -30, -15]
ANCHOR_SCALE_FACTORS = None
USE_CENTER_OFFSET = True
METHOD = 'R'
USE_ANGLE_COND = False
# --------------------------------------------RPN config
SHARE_NET = True
USE_P5 = True
IOU_POSITIVE_THRESHOLD = 0.5
IOU_NEGATIVE_THRESHOLD = 0.4

NMS = True
NMS_IOU_THRESHOLD = 0.5
MAXIMUM_DETECTIONS = 100
FILTERED_SCORE = 0.05
VIS_SCORE = 0.4
# --------------------------------------------NAS FPN config
NUM_FPN = 0
NUM_NAS_FPN = 0
USE_RELU = True
FPN_CHANNEL = 256

**Comment by yangxue0827** [comment]
看loss已经nan了 @liugaolian

**Comment by liugaolian** [comment]
> 看loss已经nan了 @liugaolian

是的，对于我这样的目标，可能是什么原因呢？期待你的回复（R2CNN检测效果还可以）

**Comment by zhouzhubin** [comment]
@liugaolian 你好，你现在的可以了吗？我训练到900step就会出现这个RuntimeWarning: divide by zero encountered in log targets_dw = np.log(gt_rois[:, 2] / ex_rois[:, 2])，然后后面就全是nan了，请问你现在解决了吗？
